lms/services/grouping/service.py

```diff
@@ -203,6 +203,31 @@ def get_groups(
 
         return self._to_groupings(user, groupings, course, self.plugin.group_type)
 
+    def get_known_groupings(self, user, course):
+        """
+        Get the groupings we already know about for the provided user.
+
+        This will include the course, and any sections / groups we have seen
+        before. This does not attempt to retrieve the groups from the LMS.
+
+        This only works for direct children of the course, not children of
```

Comment:
Currently there's no more nesting as far as I know (course -> sections/groups).

Reply:
No, but I wanted to make clear that if we ever do have an additional level of nesting that this method isn't smart enough to work with it.